apps/pterostatus/ptero_status.star

```diff
@@ -0,0 +1,195 @@
+
+def render_stat(stat, v):
+    if stat == "uptime":
+        return render.Text("Up: %s" % v)
+    if stat == "cpu":
+        return render.Text("CPU: " + str(v) + " %")
+    if stat == "mem":
+        return render.Text("Mem: %sGB" % v)
+    if stat == "disk":
+        return render.Text("Disk: %sGB" % v)
+    else:
+        return render.Text("No info..")
+
+def build_anim(k, v):
+    return animation.Transformation(
+        height = 32,
+        width = 54,
+        duration = 120,
+        origin = animation.Origin(0.5, 0.5),
+        wait_for_child = True,
+        child = render_stat(k, v),
+        keyframes = [
+            animation.Keyframe(
+                percentage = 0.0,
+                transforms = [animation.Translate(0, 32)],
+                curve = "ease_in_out",
+            ),
+            animation.Keyframe(
+                percentage = 0.2,
+                transforms = [animation.Translate(0, 0)],
+                curve = "ease_in_out",
+            ),
+            animation.Keyframe(
+                percentage = 0.8,
+                transforms = [animation.Translate(0, 0)],
+                curve = "ease_in_out",
+            ),
+            animation.Keyframe(
+                percentage = 1.0,
+                transforms = [animation.Translate(0, 32)],
+                curve = "ease_in_out",
+            ),
+        ],
+    )
+
+def build_display(url, server, key):
+    name = get_details(url, server, key)
+    stats = get_status(url, server, key)
+
+    if stats["state"] != "running":
+        status = render.Box(width = 8, height = 32, color = "#b10")
+    else:
+        status = render.Box(width = 8, height = 32, color = "#0f0")
+
+    return render.Row(
+        main_align = "space_between",
+        cross_align = "center",
+        expanded = True,
+        children = [
+            render.Column(
+                main_align = "center",
+                children = [
+                    render.Row(
+                        children = [
+                            render.Marquee(
+                                width = 54,
+                                child = render.Text("%s" % name),
+                            ),
+                        ],
+                    ),
+                    render.Box(
+                        height = 1,
+                        width = 54,
+                        color = "#333",
+                    ),
+                    render.Sequence(
+                        children = [
+                            build_anim(k, v)
+                            for k, v in stats.items()[:-1]
+                        ],
+                    ),
+                ],
+            ),
+            render.Column(
+                children = [
+                    status,
+                ],
+            ),
+        ],
+    )
+
+def main(config):
+    server = config.str("server", DEV_SERVER_ID)
+    url = config.str("url", DEV_PTERO_URL)
```

Comment:
It would be great if the app could be made to produce an illustrative default graphic for the case when no configuration is provided. That would allow us to generate app previews that give users a sense of what the app will look and feel like once it's properly configured.